datafusion/common/src/dfschema.rs

```diff
@@ -861,7 +861,10 @@ impl TryFrom<SchemaRef> for DFSchema {
 
 impl From<DFSchema> for SchemaRef {
     fn from(df_schema: DFSchema) -> Self {
-        SchemaRef::new(df_schema.into())
+        Arc::new(Schema::new_with_metadata(
+            df_schema.fields().to_owned(),
+            HashMap::new(),
```

Comment:
This change effectively discards the metadata when converting from DFSchema --> Schema (HashMap::new() is empty metadata)

So I am not surprised this can trigger the issue described (as it introduces a bug)

Reply:
Yeah, this is why I wasn't pushing code yesterday. Knew I was brain fried. Sorry about that.

Reply:
no worries -- I think we are making progress. I am particularly stoked about the `metadata.slt` idea to try and trigger some of the errors we are seeing (and I think that would make it easier to avoid this kind of regression going forward)